File: 0027-remove-element/0027-remove-element.go
/*
func removeElement(nums []int, val int) int {
    left, right := 0, len(nums)-1
    for left <= right {
        if nums[left] == val {
            nums[left], nums[right] = nums[right], nums[left]
            right--
        } else {
            left++
        }
    }
    return left
}
*/

func removeElement(nums []int, val int) int {
    i := 0
    for _, n := range nums {
        if n != val {
            nums[i] = n
            i++
        }
    }
    return i
}

File: 0027-remove-element/0027-remove-element.py
class Solution:
    def removeElement(self, nums: List[int], val: int) -> int:
        if len(nums) == 1:
            if nums[0] == val:
                return 0
            return 1
        
        left, right = 0, len(nums)-1
        while left < right:
            while 0 < right and nums[right] == val:
                right -= 1
            while left < len(nums) and nums[left] != val:
                left += 1
            if left < right:
                nums[left], nums[right] = nums[right], nums[left]
                
        print(left)        
        return left